### PR TITLE
Adding tutorials folder with example notebooks 

### PR DESCRIPTION
**Pull request by zohimchandani**
### Description

Adding notebook examples into a separate tutorials folder in python

### CODE REVIEW

**Comment by zohimchandani** [comment]
Do we know whats failing here? @bettinaheim @amccaskey

**Comment by amccaskey** [comment]
> Do we know whats failing here? @bettinaheim @amccaskey

The CI environment does not have pandoc installed. My suggestion would be to add pandoc to that devdeps Dockerfile.

**Comment by zohimchandani** [comment]
@bettinaheim this is failing as pandoc is missing even when I have added it to the devdeps.docker file. Can you check what image is being used to test this and advise further please?

**Comment by bmhowe23** [comment]
According to the [CI logs](https://github.com/NVIDIA/cuda-quantum/actions/runs/6088974591/job/16520771487?pr=560), Job `Load dependencies (amd64, llvm) / Caching` Step `Build cuda-quantum-devdeps image` produced a Docker image w/ `Digest=sha256:16485430df68cf2f9170f3ffb1b590be5791771ac6e5ab708d1c676e703fd0fe` and `ImageID=sha256:481879eb57e4a2d40e1059d023750e5bd5b320ea665ac493a3d3d6988920cade`. Based on the logs, I'm reasonably confident this had pandoc correctly installed (thanks to the updates to docker/build/devdeps.Dockerfile), but I cannot see how that is supposed to get into the job for `Create Docker images (amd64) / Documentation`. That docker _digest_ was never referenced again in the downstream jobs (and I cannot tell if it should be).

However, the _ImageID_ was later referenced in 2 interesting ways in the `Create Docker images (amd64)  cuda-quantum-dev (debug)` job:
```
./Create Docker images (amd64)  cuda-quantum-dev (debug)/7_Load prerequisites.txt:  67: 2023-09-05T19:44:50.0465680Z Deleted: sha256:481879eb57e4a2d40e1059d023750e5bd5b320ea665ac493a3d3d6988920cade
```
^^^ Why would it be deleting this ImageID before the `Documentation` has run?

And other interesting reference:

```
./Create Docker images (amd64)  cuda-quantum-dev (debug)/28_Stop containers.txt:   57:  2023-09-05T19:54:19.6823662Z  time="2023-09-05T19:44:48.635175194Z" level=error msg="response completed with error" err.code="blob unknown" err.detail=sha256:481879eb57e4a2d40e1059d023750e5bd5b320ea665ac493a3d3d6988920cade err.message="blob unknown to registry" go.version=go1.19.9 http.request.host="localhost:5000" http.request.id=6f87b97a-4200-4857-bb6e-6cc2cac61c44 http.request.method=HEAD http.request.remoteaddr="172.18.0.1:50632" http.request.uri="/v2/nvidia/cuda-quantum-devdeps/blobs/sha256:481879eb57e4a2d40e1059d023750e5bd5b320ea665ac493a3d3d6988920cade" http.request.useragent="docker/24.0.5 go/go1.20.6 git-commit/a61e2b4 kernel/5.10.186-179.751.amzn2.x86_64 os/linux arch/amd64 UpstreamClient(Docker-Client/20.10.23 \(linux\))" http.response.contenttype="application/json; charset=utf-8" http.response.duration=1.446534ms http.response.status=404 http.response.written=157 vars.digest="sha256:481879eb57e4a2d40e1059d023750e5bd5b320ea665ac493a3d3d6988920cade" vars.name="nvidia/cuda-quantum-devdeps"
```
^^^ Part of the scripts thought this container should be running and therefore stopped when stopping all the containers.

Perhaps @bettinaheim can comment on if the new Docker image that was created by this job was supposed to percolate into the GH caches via `cache/save` and `cache/restore` or if they were only supposed to stay in the registry and then get `docker pull`ed by steps.

**Comment by bettinaheim** [comment]
@zohimchandani What do you need pandoc for?
What you are installing is not actually the pandoc engine: see https://stackoverflow.com/a/71585691. You can see the same failure locally.

File: docker/build/devdeps.Dockerfile [hunk 1]
```diff
@@ -153,8 +153,8 @@ COPY --from=doxygenbuild /usr/local/bin/doxygen /usr/local/bin/doxygen
 ENV PATH="${PATH}:/usr/local/bin"
 RUN apt-get update && apt-get install -y --no-install-recommends python3 python3-pip \
```
Comment (bettinaheim) [inline docker/build/devdeps.Dockerfile:154]:
```suggestion
RUN apt-get update && apt-get install -y --no-install-recommends python3 python3-pip pandoc \
```

**Comment by bettinaheim** [comment]
@zohimchandani Please pull in this change: https://github.com/zohimchandani/cuda-quantum/pull/2

File: docs/sphinx/using/tutorials.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+*************************
+CUDA Quantum Tutorials
+*************************
+
+Tutorials that give an in depth view of CUDA Quantum and its applications in Python.
+
+.. nbgallery::
+   /examples/python/tutorials/executing_circuits.ipynb
```
Comment (bettinaheim) [inline docs/sphinx/using/tutorials.rst:8]: I think you should simply be able to indent these 4 or more whitespaces for the spellcheck errors to go away.

**Review by bettinaheim — APPROVED** [review]

**Pull request merged**